src/api/offers.js

```diff
@@ -16,23 +17,31 @@ const validateMiddleware = (req, res, next) => {
 export default ({offerService}) => {
   const router = new Router()
 
-  router.get('/getAll', (req, res) => {
-    res.json(offerService.getAllOffers())
+  router.get('/getAll', async (req, res) => {
+    try {
+      const queryResult = await offerService.getAllOffers()
+      return res.json(queryResult)
+    } catch (e) {
+      return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
+    }
   })
 
-  router.get('/', (req, res) => {
-    res.json(offerService
-      .getActiveOffers(req.city)
-      .map(offer => offer.formData))
+  router.get('/', async (req, res) => {
+    try {
+      const queryResult = await offerService.getActiveOffers(req.city)
+      return res.json(queryResult)
+    } catch (e) {
+      return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
```

Comment:
:warning: This does leak internal errors and stack traces afaik. This should not happend. This error can be logged to the stdout but shouldn't be exposed to the clients.

Reply:
Yup that's also Dev-only, I'll create a task to remove these

Reply:
With mongoose comes the cool option to set custom error messages for every item in a form. I already partially implemented this, but I think they are not returned atm. We could also return these messages here.

Reply:
I added this to a task.